src/content.js

```diff
@@ -1,244 +1,337 @@
-        )
-    );
+    titleEl.innerHTML = titleHTMLContent(title, ticketNumber);
 
     //Open up a handle for data
-    $('#partial-discussion-header').append(
-        '<div id="insertedJiraData" class="gh-header-meta">Loading ticket '+ticketNumber+'...</div>'
-    );
+    const loadingElement = buildLoadingElement(ticketNumber);
+    partialDiscussionHeaderEl.appendChild(loadingElement);
 
     //Load up data from jira
-    chrome.runtime.sendMessage(
-        {query: 'getTicketInfo', jiraUrl: jiraUrl, ticketNumber: ticketNumber},
-        function(result) {
-            const reporter = result.fields.reporter;
-            const assignee = result.fields.assignee || false;
-            let assigneeDetails = '';  
-
-            if (assignee) {
-                assigneeDetails = ` and assigned to <span class="author text-bold">
-                    <img src="${assignee.avatarUrls['16x16']}" width="16"/>
-                    ${assignee.displayName}
-                </span>`;
-            }
-
-            $("#insertedJiraData").html(
-                `<div class="TableObject gh-header-meta">
-                    <div class="TableObject-item">
-                        <span class="State State--green" style="background-color: rgb(150, 198, 222);">
-                            <img height="16" class="octicon" width="12" aria-hidden="true" src="${jiraLogo}"/>
-                            <a style="color:white;" href="${ticketUrl}" target="_blank">Jira</a>
-                        </span>
-                    </div>
-                    <div class="TableObject-item">
-                        <span class="State State--white" style="background-color: rgb(220, 220, 220);color:rgb(40,40,40);">
-                            <img height="16" class="octicon" width="12" aria-hidden="true" src="${result.fields.status.iconUrl}"/>
-                            ${result.fields.status.name}
-                        </span>
-                    </div>
-                    <div class="TableObject-item TableObject-item--primary">
-                        <strong>
-                            <a href="${ticketUrl}" target="_blank">
-                                [${ticketNumber}] - ${result.fields.summary}
-                            </a>
-                        </strong> - Reported by <span class="author text-bold">
-                            <img src="${reporter.avatarUrls['16x16']}" width="16"/>
-                                ${reporter.displayName}
-                        </span>${assigneeDetails}
-                    </div>
-                </div>`
-            );
+    try {
+        const result = await sendMessage({ query: 'getTicketInfo', jiraUrl, ticketNumber })
+        if (result.errors) {
+            throw new Error(result.errorMessages);
         }
-    );
+        loadingElement.innerHTML = headerBlock(ticketNumber, result.fields);
+    } catch(e) {
+        console.error('Issue fetching data', e)
```

Comment:
Might want to rename "Issue" to "Error" or "Problem", as "issue" is kind of ambiguous when we're talking about "issues in Jira" :stuck_out_tongue_winking_eye:

Reply:
Ha! No joke.